projects/packages/my-jetpack/src/products/class-search.php

```diff
@@ -155,11 +155,21 @@ public static function get_wpcom_free_product_slug() {
 	}
 
 	/**
-	 * Returns true if the new_pricing_202208 is set to not empty in URL for testing purpose.
+	 * Returns true if the new_pricing_202208 is set to not empty in URL for testing purpose, or it's active.
 	 */
 	public static function is_new_pricing_202208() {
 		// phpcs:ignore WordPress.Security.NonceVerification.Recommended, WordPress.Security.ValidatedSanitizedInput.InputNotSanitized, WordPress.Security.ValidatedSanitizedInput.MissingUnslash
-		return isset( $_GET['new_pricing_202208'] ) && $_GET['new_pricing_202208'];
+		if ( isset( $_GET['new_pricing_202208'] ) && $_GET['new_pricing_202208'] ) {
+			return true;
+		}
+
+		$record_count   = intval( Search_Stats::estimate_count() );
+		$search_pricing = static::get_pricing_from_wpcom( $record_count );
```

Comment:
Note that we aren't concerned with `$record_count` here, but since we use it in the cache key, I hope that we just won't have to do the API call at all that way.